pennylane_qiskit/qiskit_device.py

```diff
@@ -12,124 +12,204 @@
-    "CRZ": lib.CRZGate,
-    "PhaseShift": lib.PhaseGate,
-    "QubitStateVector": lib.Initialize,
-    "StatePrep": lib.Initialize,
-    "Toffoli": lib.CCXGate,
-    "QubitUnitary": lib.UnitaryGate,
-    "U1": lib.U1Gate,
-    "U2": lib.U2Gate,
-    "U3": lib.U3Gate,
-    "IsingZZ": lib.RZZGate,
-    "IsingYY": lib.RYYGate,
-    "IsingXX": lib.RXXGate,
-    "S": lib.SGate,
-    "T": lib.TGate,
-    "SX": lib.SXGate,
-    "Adjoint(S)": lib.SdgGate,
-    "Adjoint(T)": lib.TdgGate,
-    "Adjoint(SX)": lib.SXdgGate,
-    "CY": lib.CYGate,
-    "CH": lib.CHGate,
-    "CPhase": lib.CPhaseGate,
-    "CCZ": lib.CCZGate,
-    "ECR": lib.ECRGate,
-    "Barrier": lib.Barrier,
-    "Adjoint(GlobalPhase)": lib.GlobalPhaseGate,
-}
-
-
-def _get_backend_name(backend):
+        import pennylane as qml
+        from pennylane_qiskit import qiskit_session
+        from qiskit_ibm_runtime import QiskitRuntimeService
+
+        # get backend
+        service = QiskitRuntimeService(channel="ibm_quantum")
+        backend = service.least_busy(simulator=False, operational=True)
+
+        # initialize device
+        dev = qml.device('qiskit.remote', wires=2, backend=backend)
+
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RX(x, 0)
+            qml.CNOT([0, 1])
+            return qml.expval(qml.PauliZ(1))
+
+        angle = 0.1
+
+        with qiskit_session(dev) as session:
+
+            res = circuit(angle)[0] # you queue for the first execution
+
+            # then this loop executes immediately after without queueing again
+            while res > 0:
+                angle += 0.3
+                res = circuit(angle)[0]
+    """
+    # Code to acquire session:
+    existing_session = device._session
+    session = Session(backend=device.backend)
+    device._session = session
     try:
-        return backend.name()  # BackendV1
-    except TypeError:  # pragma: no cover
-        return backend.name  # BackendV2
+        yield session
+    finally:
+        # Code to release session:
+        session.close()
+        device._session = existing_session
+
+
+def accepted_sample_measurement(m: qml.measurements.MeasurementProcess) -> bool:
+    """Specifies whether or not a measurement is accepted when sampling."""
+
+    return isinstance(
+        m,
+        (
+            qml.measurements.SampleMeasurement,
+            qml.measurements.ClassicalShadowMP,
+            qml.measurements.ShadowExpvalMP,
+        ),
+    )
 
 
-class QiskitDevice(QubitDevice, abc.ABC):
-    r"""Abstract Qiskit device for PennyLane.
+@transform
+def split_execution_types(
+    tape: qml.tape.QuantumTape,
+) -> (Sequence[qml.tape.QuantumTape], Callable):
+    """Split into separate tapes based on measurement type. Counts and sample-based measurements
+    will use the Qiskit Sampler. ExpectationValue and Variance will use the Estimator, except
+    when the measured observable does not have a `pauli_rep`. In that case, the Sampler will be
```

Comment:
When do we not have a `pauli_rep` for the observables? 🤔 
For `Projectors` and `Hermitian` unitaries, can we not rely on decomposing them to a Pauli basis either via PL or qiskit and then use them with `Estimator`? For the rest, if this fails the fallback method can still be the `Sampler` approach.

Reply:
For example, observables like `qml.Hadamard` don't have a `pauli_rep`. In this case we use the `Sampler` instead and diagonalize ourselves. 

>For Projectors and Hermitian unitaries, can we not rely on decomposing them to a Pauli basis either via PL or qiskit and then use them with Estimator? 

I _think_ this is doable but is the extra work worth it when we could just use the Sampler approach instead?